addons/product/models/product_pricelist_item.py

```diff
@@ -460,3 +460,11 @@ def _compute_price_before_discount(self, *args, **kwargs):
             pricelist_rule = pricelist_item
 
         return pricelist_rule._compute_base_price(*args, **kwargs)
+
+    @api.constrains('base_pricelist_id')
+    def _check_company_pricelist_in_multi_company_context(self):
+        for record in self:
+            if (record.base_pricelist_id.company_id
+                and record.base_pricelist_id.company_id != record.pricelist_id.company_id):
```

Comment:
Though for the if statement, this works fine too:
```suggestion
            if record.base_pricelist_id.company_id \
               and record.base_pricelist_id.company_id != record.pricelist_id.company_id:
```